### ISSUE TITLE
Details about "Grade Conversion Information"

### ISSUE THREAD

**Issue by wrygiel** [erasmus-without-paper/ewp-wp2-use-cases#2]
@StefanJahnke, could you please provide more details about use case 6?

> The grade conversion information will be transferred from the Receiving Institution to the Sending Institution.

Could you explain what exactly do you mean by _grade conversion information_? The description of this use case is a little too broad for me to address properly. I'm not sure if what you need is included in the EMREX ELMO file we intend to reuse in our ToR APIs.

**Comment by StefanJahnke** [comment]
Hello there,
I am not sure what information is used in EMREX. 

The ECTS Users' guide and also the Learning Agreement guidelines clearly state that the information about grade distribution at the receiving institution should be provided, so that the sending institution can convert grades given at the receiving institution into local grades.

Our colleagues Valere and Anthony have been highly committed to this and created the [EGRACONS](http://egracons.eu/) project, which does automatic conversion once the data is entered into their system.

During our workshop, the colleagues argued that we should have a simple API that would allow the transfer of those information. We now have to define how the information looks like and I will try to finalise the data dictionary by the end of the week. It will most likely be based on the template that the EGRACONS project is providing.

The EGRACONS should probably also be the first use-case where an intermediate connector could be implemented. The use-case would look like this: EGRACONS receives the Learning Agreement data and the grade conversion data and can automatically convert grades for the sending institution.

**Comment by wrygiel** [comment]
> We now have to define how the information looks like and I will try to finalise the data dictionary by the end of the week.

Okay! Please respond here when you know more.

From WP4's perspective the important question is:
- Do you require this information on the _institution_ level? (Institutions API)
- On a _course_ level? (Courses API)
- On a _course + academic term_ level? (Transcript of Records API)

As far as I remember, EMREX uses the third option - a grade distribution table can be included in the ToR file. However, I'm not sure if this constitutes proper "grade conversion information".

@erasmus-without-paper/wp3 will also be interested in the subject.

**Comment by wrygiel** [comment]
As agreed in Warsaw, [UC6](https://github.com/erasmus-without-paper/ewp-wp2-use-cases/tree/332802134978d803a8c0423b00f823c6c3a5715f#6-grade-conversion-uc) needs to be modified because the current requirement stated there will not be handled by the EWP network.

@erasmus-without-paper/wp2

**Comment by StefanJahnke** [comment]
Hello Wojtek,
would you like me to change the diagram?

**Comment by wrygiel** [comment]
Either change, or remove - I guess it doesn't matter much.

**Comment by StefanJahnke** [comment]
Could you elaborate on what exactly you decided on? 

We can of course completely remove the use-case but the way I understand is that you want to create an EGRACON module?

**Comment by wrygiel** [comment]
I believe @jpbacelar will be able to explain it better.

The way I understood it, UC6 was about integration with EGRACONS, and this cannot be accomplished because EGRACONS doesn't expose any API. In other words, UC6 was **not** about extending EWP - it was about extending EGRACONS, so it's out of our scope.

**Comment by wrygiel** [comment]
I think I finally understood what UC6 expects EWP to do. The original question I have asked to @erasmus-without-paper/wp2 was:

> Could you explain what exactly do you mean by grade conversion information? The description of this use case is a little too broad for me to address properly.

I will quote from [here](https://github.com/erasmus-without-paper/ewp-wp3-data-model/issues/5):

@StefanJahnke wrote:

> We agreed early on that we should try to enable the EGRACONS project to make usage of the EWP and integrate it somehow.

@janinamincer-daszkiewicz wrote:

> Grade conversion might be based on statistical distribution of grades (...) Grade distribution is needed for the proper grade conversion.

In other words, UC6 wants EWP to supply **statistical distribution of grades**.

Followup question:
- Is it supposed to be per institution and per grade scale? Or, per course? Or both?

BTW, @kaiqu might also be interested.

**Comment by StefanJahnke** [comment]
Before writing long paragraphs trying to mirror information and probably failing or only giving partically correct information, I belive that a relatively comprehensive and complete information on what grade conversion and grading tables are (that's the data that should be transferable via EWP), I would refer to this link:
http://egracons.eu/page/about

**Comment by kaiqu** [comment]
> I belive that a relatively comprehensive and complete information on what grade conversion and grading tables are (that's the data that should be transferable via EWP), I would refer to this link:
> http://egracons.eu/page/about

I think the best way to model grade distribution is on the lowest level, i.e. LOI. Aggregations to LOS and HEI are just calculations from there. This is included in the coming version of the model.

**Comment by wrygiel** [comment]
> I think the best way to model grade distribution is on the lowest level, i.e. LOI. Aggregations to LOS and HEI are just calculations from there.

In theory, you're right of course. However, providing more information than is "strictly necessary", may sometimes grow to become a privacy issue... (I don't see a real privacy issue here, it's just my instinct talking.)

Also, this would require a separate API, because we don't have an API for LOIs. This means that we would need to have LOI IDs, and also a way to find/index these LOI IDs. We would probably also need unique identifiers of different grading scales.

To sum up, having this on a LOI level might mean much work for everyone, so we shouldn't make too hasty decisions, I think.

**Comment by kaiqu** [comment]
> > I think the best way to model grade distribution is on the lowest level, i.e. LOI. Aggregations to LOS and HEI are just calculations from there.
> 
> In theory, you're right of course. However, providing more information than is "strictly necessary", may sometimes grow to become a privacy issue [...]
> 
> Also, this would require a separate API, because we don't have an API for LOIs. This means that we would need to have LOI IDs [...]

It's important to differentiate between the model and the APIs:

As far as I understand, the grade distribution is supposed to be returned in the ToRs API, which does involve LOI. If there is no distribution stored on this level, but it's meaningful to deliver a more aggregated distribution, the API can return this instead.

E.g:
- A ToRs response concerns a Course LOI (i.e. belonging to a LOS of type Course)
- The LOS is part of a higher LOS (e.g. of type Programme)
- There is no grade distribution stored at the Course x Academic Term (LOI) level
- But there is one at the Programme x Academic Term level
- Or at the HEI x Academic Term level

> To sum up, having this on a LOI level might mean much work for everyone, so we shouldn't make too hasty decisions, I think.

And I think not :-)

**Comment by wrygiel** [comment]
> As far as I understand, the grade distribution is supposed to be returned in the ToRs API, which does involve LOI.

ToRs are accessible to the sending HEI **only**, and they contain only a very limited subset of courses.

I believe @erasmus-without-paper/wp2 wants us to design a way of providing grade distribution:
- **directly** for EGRACONS (meaning - EGRACONS would have a direct read access to those),
- with much better coverage (much broader sets of courses, not necessarily related to any mobility).

**Comment by janinamincer-daszkiewicz** [comment]
There are two possible ways of dealing with grade conversion:
1. EGRACON style
2. Swedish or Norwegian (?) style where for every course you deliver grade distribution
EGRACON style is more difficult and more controversial (my IRO is a little bit sceptical about using it in transcript of records for short-term studies). Accepting grade distribution for a course is straightforward, some partners (Sweden or Norway?) already deliver it. I think that we should at least support (2).
Supporting (1) would of course be nice, at least at the API level - even if we will not be able to implement it during EWP-1.

**Comment by wrygiel** [comment]
> I think that we should at least support (2).

I think you mean including grade distribution in ToRs, as Kai proposed - right?

True, this would be much easier for us to design. Designing a dedicated solution for EGRACONS would probably be much harder, because we don't have contact with EGRACONS programmers.

**Comment by janinamincer-daszkiewicz** [comment]
> I think you mean including grade distribution in ToRs, as Kai proposed - right?

Right

**Comment by kaiqu** [comment]
> There are two possible ways of dealing with grade conversion:
> 1. EGRACON style
> 2. Swedish or Norwegian (?) style where for every course you deliver grade distribution
> 
> EGRACON style is more difficult and more controversial (my IRO is a little bit sceptical about using it in transcript of records for short-term studies). Accepting grade distribution for a course is straightforward, some partners (Sweden or Norway?) already deliver it. I think that we should at least support (2).

So - do we have a conclusion to this? If we go for (2), it is already modeled in the current version.

**Comment by janinamincer-daszkiewicz** [comment]
WP2 should decide. In my opinion it would be nice if we would model it, even knowing that it will not be implemented as part of EWP-1. There is a nice (not too long) description in http://europass.cedefop.europa.eu/sites/default/files/ectsusersguide_pdf.pdf
Annex 3, page 41.

**Comment by kaiqu** [comment]
> > So - do we have a conclusion to this? If we go for (2), it is already modeled in the current version.
> 
> WP2 should decide. In my opinion it would be nice if we would model it, even knowing that it will not be implemented as part of EWP-1.

Ok, I'll leave it in the model for the time being.

> There is a nice (not too long) description in
> http://europass.cedefop.europa.eu/sites/default/files/ectsusersguide_pdf.pdf
> Annex 3, page 41.

Thanks, I'll have a look.

**Comment by StefanJahnke** [comment]
I think it's difficult for me to decide on behalf of the consortium on how
grade distribution should be implemented. If I remember correctly, EGRACONS
is using a certain ISCED level to "summarise" grading tables. Shouldn't it
be possible to convert single course distribution, if one knows which ISCED
code they fall under? It would then be up to EGRACONS to implement a
"connector" that could do automatic summarising and convert grades. I would
be happy if you could bring this up especially with Valere and Anthony, as
they are the ones standing behind the EGRACONS project.

**Comment by wrygiel** [comment]
> Also, this would require a separate API, because we don't have an API for LOIs.

During my work on the Courses API I have determined that LOIs can (and should) be [introduced there](https://github.com/erasmus-without-paper/ewp-specs-api-courses/blob/v0.2.0/response.xsd#L209).

> This means that we would need to have LOI IDs, and also a way to find/index these LOI IDs.

BTW, these also were [introduced](https://github.com/erasmus-without-paper/ewp-specs-api-courses/blob/v0.2.0/response.xsd#L220). (Although there's currently no option to access LOIs directly by their IDs.)

> We would probably also need unique identifiers of different grading scales.

We still don't have those - we have [labels and descriptions](https://github.com/erasmus-without-paper/ewp-specs-api-courses/blob/v0.2.0/response.xsd#L268) only. However, as discussed [here](https://github.com/erasmus-without-paper/ewp-specs-api-courses/issues/4) we have managed to include [basic grade conversion](https://github.com/erasmus-without-paper/ewp-specs-api-courses/blob/v0.2.0/response.xsd#L303) information on the LOI level too.

Closing this.
